pocketimdb/movies/views.py

```diff
@@ -19,39 +19,22 @@ class MovieViewSet(mixins.ListModelMixin,
-            dislikes=Coalesce(Count('movie_likes__like', filter=Q(movie_likes__like=MovieLike.Like.DISLIKE)), 0),
+            likes=Coalesce(Count('movie_likes__like', filter=Q(movie_likes__like=Like.LIKE)), 0),
+            dislikes=Coalesce(Count('movie_likes__like', filter=Q(movie_likes__like=Like.DISLIKE)), 0),
             user_liked_or_disliked=Coalesce(Sum('movie_likes__like', filter=Q(movie_likes__user=self.request.user)), 0),
         ).order_by('id')
 
-    @action(detail=True, url_path='likes')
+    @action(methods=['POST', 'DELETE'], detail=True, url_path='likes')
     def like(self, request, pk):
-        pass
-
-    @like.mapping.post
-    def add_like(self, request, pk):
-        serializer = AddMovieLikeSerializer(data=request.data)
-        serializer.is_valid(raise_exception=True)
-        if Movie.objects.filter(pk=pk, movie_likes__user=request.user).exists():
-            error = {"unique_together_error": ["Fields movie and user must make a unique set."]}
-            return Response(error, status=HTTP_403_FORBIDDEN)
-        MovieLike.objects.create(**serializer.data, movie_id=pk, user=request.user)
-        return Response(status=HTTP_201_CREATED)
-
-    @like.mapping.patch
-    def flip_like(self, request, pk):
-        movie_likes = MovieLike.objects.filter(movie_id=pk, user=request.user)
-        if len(movie_likes) == 0:
-            error = {"not_exists_error": ["There is no like/dislike for chosen movie and user."]}
-            return Response(error, status=HTTP_400_BAD_REQUEST)    
-        movie_likes.update(like=Case(When(like=MovieLike.Like.LIKE, then=MovieLike.Like.DISLIKE), default=MovieLike.Like.LIKE))
-        return Response(status=HTTP_200_OK)
-
-    @like.mapping.delete
-    def remove_like(self, request, pk):
-        movie_likes = MovieLike.objects.filter(movie_id=pk, user=request.user)
-        if len(movie_likes) == 0:
-            error = {"not_exists_error": ["There is no like/dislike for chosen movie and user."]}
-            return Response(error, status=HTTP_400_BAD_REQUEST)    
-        movie_likes.delete()
-        return Response(status=HTTP_204_NO_CONTENT)
\ No newline at end of file
+        if (request.method == 'POST'):
+            serializer = AddMovieLikeSerializer(data=request.data)
+            serializer.is_valid(raise_exception=True)
+            MovieLike.objects.update_or_create(movie_id=pk, user=request.user, defaults={**serializer.data})
+            return Response(status=HTTP_200_OK)
+        else:
+            movie_likes = MovieLike.objects.filter(movie_id=pk, user=request.user)
+            if not movie_likes.exists():
+                error = {"not_exists_error": ["There is no like/dislike for chosen movie and user."]}
+                return Response(error, status=HTTP_404_NOT_FOUND)    
+            movie_likes.delete()
+            return Response(status=HTTP_204_NO_CONTENT)
```

Comment:
nemoj ovo da radis, napravi dvije posebne metode, kao sto reko treba izbjegavati if/else ako je to moguce i da budemo sto eksplicitniji, jer ovako mene ako zanima post ili delete moram proci kroz citavu akciju da vidim koja je razlika u sta se na sta odnosi, tako da budi sto eksplicitniji kod ovoga